lib/Console/Traits/Parallelization.php

```diff
@@ -103,4 +108,191 @@ protected function getContainer()
+
+        $numberOfProcessesDefined = null !== $input->getOption('processes');
+        $numberOfProcesses = $numberOfProcessesDefined ? (int) $input->getOption('processes') : 1;
+        $hasItem = (bool) $input->getArgument('item');
+        $items = $hasItem ? [$input->getArgument('item')] : $this->fetchItems($input);
+        $count = count($items);
+        $segmentSize = 1 === $numberOfProcesses && !$numberOfProcessesDefined ? $count : $this->getSegmentSize();
+        $batchSize = $this->getBatchSize();
+        $rounds = 1 === $numberOfProcesses ? 1 : ceil($count * 1.0 / $segmentSize);
+        $batches = ceil($segmentSize * 1.0 / $batchSize) * $rounds;
+
+        Assert::greaterThan(
+            $numberOfProcesses,
+            0,
+            sprintf(
+                'Requires at least one process. Got "%s"',
+                $input->getOption('processes')
+            )
+        );
+
+        if (!$hasItem && 1 !== $numberOfProcesses) {
+            // Shouldn't check this when only one item has been specified or
+            // when no child processes is used
+            Assert::greaterThanEq(
+                $segmentSize,
+                $batchSize,
+                sprintf(
+                    'The segment size should always be greater or equal to '
+                    .'the batch size. Got respectively "%d" and "%d"',
+                    $segmentSize,
+                    $batchSize
+                )
+            );
+        }
+
+        $output->writeln(sprintf(
+            'Processing %d %s in segments of %d, batches of %d, %d %s, %d %s in %d %s',
+            $count,
+            $this->getItemName($count),
+            $segmentSize,
+            $batchSize,
+            $rounds,
+            1 === $rounds ? 'round' : 'rounds',
+            $batches,
+            1 === $batches ? 'batch' : 'batches',
+            $numberOfProcesses,
+            1 === $numberOfProcesses ? 'process' : 'processes'
+        ));
+        $output->writeln('');
+
+        $progressBar = new ProgressBar($output, $count);
+        $progressBar->setFormat('debug');
+        $progressBar->start();
+
+        if ($count <= $segmentSize || (1 === $numberOfProcesses && !$numberOfProcessesDefined)) {
+            // Run in the master process
+
+            $itemsChunks = array_chunk(
+                $items,
+                $this->getBatchSize(),
+                false
+            );
+
+            foreach ($itemsChunks as $items) {
+                $this->runBeforeBatch($input, $output, $items);
+
+                foreach ($items as $item) {
+                    $this->runTolerantSingleCommand((string) $item, $input, $output);
+
+                    $progressBar->advance();
+                }
+
+                $this->runAfterBatch($input, $output, $items);
+            }
+        } else {
+            // Distribute if we have multiple segments
+            $consolePath = realpath(getcwd().'/bin/console');
```

Comment:
@andreas-gruenwald 

Why is current working directory assumed? In my test it failed when the invocation of the master process (calling `bin/console`) happened outside of the project root, with `realpath()` returning `false`.

```
In Assert.php line 2220:
 
  [InvalidArgumentException]
  Expected a string. Got: boolean
 

Exception trace:
  at /var/www/share/REDUCTED/pimcore/htdocs/vendor/webmozart/assert/src/Assert.php:2220
 Webmozart\Assert\Assert::reportInvalidArgument() at /var/www/share/REDUCTED/pimcore/htdocs/vendor/webmozart/assert/src/Assert.php:233
 Webmozart\Assert\Assert::string() at /var/www/share/REDUCTED/pimcore/htdocs/vendor/webmozart/assert/src/Assert.php:1592
 Webmozart\Assert\Assert::fileExists() at /var/www/share/REDUCTED/pimcore/htdocs/vendor/pimcore/pimcore/lib/Console/Traits/Parallelization.php:252
 Pimcore\Bundle\EcommerceFrameworkBundle\Command\IndexService\ProcessUpdateIndexQueueCommand->executeMasterProcess() at /var/www/share/REDUCTED/pimcore/htdocs/vendor/webmozarts/console-parallelization/src/Parallelization.php:287
```

Reply:
This is a known issue.
See https://github.com/webmozarts/console-parallelization/issues/32.

You are right, we should find a solution for that. I am going to take a look.

Reply:
I just created a PR for that.

Reply:
@fashxp It's now possible to override the console path, for pimcore, if the problem is still existing: https://github.com/webmozarts/console-parallelization/pull/43